### PR TITLE
[Build] fix: missing nvcc flags when compiling with unittests

### PR DESCRIPTION
**Pull request by gedoensmax**
When configured using the following CMake ops Clion is not able to configure due to checking with `nvcc ... --dryrun tmp.cu`:
```
cmake -G Ninja -Donnxruntime_USE_TENSORRT="ON" -Donnxruntime_USE_CUDA="ON" -Donnxruntime_USE_CUDA_NHWC_OPS="ON" -DCMAKE_CUDA_ARCHITECTURES="native" -Donnxruntime_NVCC_THREADS=1 -Donnxruntime_ENABLE_NVTX_PROFILE="ON" -Donnxruntime_USE_TENSORRT_BUILTIN_PARSER="ON" -DCMAKE_CUDA_COMPILER_LAUNCHER="ccache" -Donnxruntime_BUILD_UNIT_TESTS="ON" -Donnxruntime_USE_TRITON_KERNEL=OFF -Donnxruntime_USE_FLASH_ATTENTION=OFF
```
Without building the unittests everything works fine. I believe my changes only follow the logic that is actually desired. If `NVCC_HAS_STRICT_ALIASING` is set to false it should not be possible to add this as a CUDA flag. Same is true for `HAS_NOERROR` as seen in `adjust_global_compile_flags.cmake`

@hariharans29 can you help review this?

### CODE REVIEW

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -984,9 +984,12 @@ function(onnxruntime_set_compile_flags target_name)
       foreach(FLAG ${ORT_WARNING_FLAGS})
         target_compile_options(${target_name} PRIVATE "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:--compiler-options ${FLAG}>")
       endforeach()
-      if ((NVCC_HAS_STRICT_ALIASING AND "${target_name}" MATCHES "cuda") OR (HAS_STRICT_ALIASING AND NOT "${target_name}" MATCHES "cuda"))
+      if (${NVCC_HAS_STRICT_ALIASING} AND "${target_name}" MATCHES "cuda")
```
Comment (tianleiwu) [inline cmake/CMakeLists.txt:987]: There is build error, use original naming?
${NVCC_HAS_STRICT_ALIASING} => NVCC_HAS_STRICT_ALIASING
${HAS_STRICT_ALIASING} => HAS_STRICT_ALIASING

Comment (gedoensmax) [reply cmake/CMakeLists.txt]: Done I was not aware that it makes a big change. I though those "variable evaluation brackets" would actually be safer.

**Review by tianleiwu — APPROVED** [review]

**Pull request merged**